cmd/edenPod.go

```diff
@@ -0,0 +1,546 @@
+			},
+		},
+		Dns: nil,
+	}
+	return netInst, nil
+}
+
+func checkAppInstanceConfig(app *config.AppInstanceConfig, appName string, appType string, appUrl string, appVersion string) bool {
+	if app == nil {
+		return false
+	}
+	if app.Displayname == appName {
+		return true
+	}
+	return false
+}
+
+func createAppInstanceConfig(img *config.Image, appName string, netInstId string, appType string, appUrl string, appVersion string) (*config.AppInstanceConfig, error) {
+	var app *config.AppInstanceConfig
+	id, err := uuid.NewV4()
+	if err != nil {
+		return nil, err
+	}
+	switch appType {
+	case "docker":
+		app = &config.AppInstanceConfig{
+			Uuidandversion: &config.UUIDandVersion{
+				Uuid:    id.String(),
+				Version: "1",
+			},
+			Fixedresources: &config.VmConfig{
+				Memory:     1024000,
+				Maxmem:     1024000,
+				Vcpus:      1,
+				Rootdev:    "/dev/xvda1",
+				Bootloader: "/usr/bin/pygrub",
+			},
+			Drives: []*config.Drive{{
+				Image: img,
+			}},
+			Activate:    true,
+			Displayname: appName,
+			Interfaces: []*config.NetworkAdapter{{
+				Name:      "default",
+				NetworkId: netInstId,
+				Acls: []*config.ACE{{
+					Matches: []*config.ACEMatch{{
+						Type: "host",
+					}},
+					Id: 1,
+				}},
+			}},
+		}
+		return app, nil
+	default:
+		return nil, fmt.Errorf("not supported appType")
+	}
+}
+
+var podDeployCmd = &cobra.Command{
+	Use:   "deploy <docker>://<TAG>[:<VERSION>]",
+	Short: "Deploy app in pod",
+	Long:  `Deploy app in pod.`,
+	Args:  cobra.ExactArgs(1),
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		assingCobraToViper(cmd)
+		_, err := utils.LoadConfigFile(configFile)
+		if err != nil {
+			return fmt.Errorf("error reading config: %s", err.Error())
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		appLink := args[0]
+		rand.Seed(time.Now().UnixNano())
+		appName := namesgenerator.GetRandomName(0)
+		if podName != "" {
+			appName = podName
+		}
+		params := getParams(appLink, defaults.DefaultPodLinkPattern)
+		if len(params) == 0 {
+			log.Fatalf("fail to parse <docker>://<TAG>[:<VERSION>] from argument (%s)", appLink)
+		}
+		appType := ""
+		appUrl := ""
+		appVersion := ""
+		ok := false
+		if appType, ok = params["TYPE"]; !ok || appType == "" {
+			log.Fatalf("cannot parse appType (not [docker]): %s", appLink)
+		}
+		if appUrl, ok = params["TAG"]; !ok || appUrl == "" {
+			log.Fatalf("cannot parse appTag: %s", appLink)
+		}
+		if appVersion, ok = params["VERSION"]; !ok || appVersion == "" {
+			log.Debugf("cannot parse appVersion from %s will use latest", appLink)
+			appVersion = "latest"
+		}
+
+		changer := &adamChanger{}
+		ctrl, dev, err := changer.getControllerAndDev()
+		if err != nil {
+			log.Fatalf("getControllerAndDev: %s", err)
+		}
+		var datastore *config.DatastoreConfig
+		for _, ds := range ctrl.ListDataStore() {
+			if checkDataStore(ds, appType, appUrl) {
+				datastore = ds
+				break
+			}
+		}
+		if datastore == nil {
```

Comment:
Can we move all verifications and logging into functions? 
Will be much better. 
I mean  

if datastore, err = createDataStore(appType, appUrl); err != nil {
				log.Fatalf("cannot create datastore: %s", err)
			}
			if err = ctrl.AddDataStore(datastore); err != nil {
				log.Fatalf("AddDataStore: %s", err)
			}
			log.Infof("new datastore created %s", datastore.Id)

becomes datastore  = createDataStore(appType, appUrl) and err is handled inside createDataStore function.